lib/chef/provider/user/useradd.rb

```diff
@@ -140,7 +140,8 @@ def update_options(field, option, opts)
 
         def useradd_options
           opts = []
-          opts << "-r" if new_resource.system
+          # Solaris doesn't support system users.
+          opts << "-r" if new_resource.system && !node.platform?("solaris2")
```

Comment:
hrmmm, i seem to have done a github review fail....   

this should really go into the user/solaris.rb file rather than special casing it here.

the useradd class here is in the process of getting abandoned and i'm going to extract the code out to solaris, aix, and linux providers that inherit only from User.   you could start by just copying useradd_options directly into the solaris provider and then changing it.

also its probably not only the solaris2 platform that needs this but all the solarisish platforms as well.